// mongo/index.js
var mongoose = require('mongoose');
var uri = 'mongodb://127.0.0.1:27017/ksaltstack';
var Schema = mongoose.Schema;

var userSchema = new Schema({
    user     : {type: String},
    password : {type: String},
    token    : {type: String},
    start    : {type: Date},
    expire   : {type: Date},
    eauth    : {type: String},
    perms    : {type: Array}
});

var serverSchema = new Schema({
    ipv4              : {type: Array},
    ipv6              : {type: Array},
    ip_interfaces     : {type: Object},
    os                : {tpye: String},
    osrelease         : {type: String},
    kernelrelease     : {type: String},
    zmqversion        : {type: String},
    host              : {type: String},
    saltversion       : {type: String},
    nodename          : {type: String},
    cpu_model         : {type: String},
    num_cpus          : {type: Number},
    osarch            : {type: String},
    locale_info       : {type: Object},
    machine_id        : {type: String},
    mem_total         : {type: Number},
    hwaddr_interfaces : {type: Object},
    ip4_interfaces    : {type: Object},
    serialnumber      : {type: String},
    master            : {type: String},
    manufacturer      : {type: String}
});

var action = {
    conn: function() {
        mongoose.connect(uri)
    },
    userModel: function() {
        return mongoose.model('users', userSchema, 'users')
    },
    serverModel: function() {
        return mongoose.model('servers', serverSchema, 'servers')
    }
};

module.exports = action;
